File: pingYouGou-manager-web/src/main/resources/static/js/service/specificationService.js
// 服务层
app.service("specificationService", function ($http, pingYouGouServer) {
    /**
     * 获取分页数据
     * @param page 当前页
     * @param size 每页数据
     * @param name 品牌名
     * @param firstChar 首字母
     */
    this.findPage = function (page, size, specName) {
        let url;
        if (specName == null) {
            url = `${pingYouGouServer}/specification/findSpecificationPage.do?pageNum=${page}&pageSize=${size}`;
        } else if (specName != null) {
            url = `${pingYouGouServer}/specification/findSpecificationPage.do?pageNum=${page}&pageSize=${size}&specName=${specName}`;
        }
        return $http({
            url: url,
            method: 'GET'
        });
    }

    // 检测输入的品牌是否已经添加
    this.checkSpecification = function (specName) {
        return $http.get(
            `${pingYouGouServer}/specification/checkSpecification.do?specName=${specName}`
        );
    }

    // 新增品牌
    this.addSpecification = function (specificationGroup) {
        return $http({
            url: `${pingYouGouServer}/specification/addSpecification.do`,
            method: "POST",
            data: {
                specification: specificationGroup.specification,
                specificationOptionList: specificationGroup.specificationOptions
            }
        });
    }

    // 删除品牌
    this.deleteSpecification = function (selectIds) {
        return $http({
            url: `${pingYouGouServer}/specification/deleteSpecification.do?selectIds=${selectIds}`,
            method: "GET"
        });
    }

    // 修改品牌
    this.modifySpecification = function (specificationId, specificationName) {
        return $http({
            url: `${pingYouGouServer}/specification/updateSpecification.do`,
            method: "POST",
            data: {
                id: specificationId,
                specName: specificationName
            }
        });
    }

    this.findAllInfo = function (specificationId) {
        return $http({
            url : `${pingYouGouServer}/specification/findAllInfo.do?id=${specificationId}`,
            method: "GET"
        });
    }

    /**
     * 查询分类JSON数据
     * @returns {*}
     */
    this.findSpecificationIdAndName = function () {
        return $http({
            url: `${pingYouGouServer}/specification/findSpecificationIdAndName.do`,
            method: "GET"
        })
    }

})

File: pingYouGou-sellerGoods-interface/src/main/java/com/lemonzuo/sellergoods/service/SpecificationService.java
package com.lemonzuo.sellergoods.service;

import com.lemonzuo.model.PageResult;
import com.lemonzuo.model.Specification;

import java.util.List;
import java.util.Map;

/**
 * @author LemonZuo
 * @create 2019-08-14 19:37
 */
public interface SpecificationService {
    /**
     * 查询分类
     * @param specification
     * @return
     */
    PageResult<Specification> findSpecification(int pageNum, int pageSize, Specification specification);

    /**
     * 查询分类是否存在
     * @param specification
     * @return
     */
    Specification checkSpecification(Specification specification);
    /**
     * 新增分类
     * @param specification
     * @return
     */
    Specification addSpecification(Specification specification);

    /**
     * 更新分类
     * @param specification
     */
    void updateSpecification(Specification specification);

    /**
     * 删除分类
     * @param id
     */
    void deleteSpecification(Long id);

    /**
     * 查询一条分类的所有信息
     * @param id
     * @return
     */
    Specification findAllInfo(Long id);

    /**
     * 返回JSON分类ID和分类名
     * @return
     */
    List<Map> findSpecificationIdAndName();
}
